tools/nxmake2cmake.py

```diff
@@ -0,0 +1,270 @@
+    raise Exception("Must be using Python 3")
+
+
+def board_cmakelist_get(path):
+    template = """# ##############################################################################
+# $board_path/CMakeLists.txt
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more contributor
+# license agreements.  See the NOTICE file distributed with this work for
+# additional information regarding copyright ownership.  The ASF licenses this
+# file to you under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License.  You may obtain a copy of
+# the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+#
+# ##############################################################################
+
+add_subdirectory(src)
+"""
+    temp_obj = Template(template)
+    return temp_obj.substitute(board_path=path)
+
+
+def add_board_cmakelist(path):
+    cmpath = os.path.join(path, "CMakeLists.txt")
+
+    print("Add", cmpath)
+
+    with open(cmpath, "w") as f:
+        content = board_cmakelist_get(path)
+        f.write(content)
+
+
+def file_update_path(s, oldpath, newpath):
+    return s.replace(oldpath, newpath)
+
+
+def convert_make_conditional(content):
+    content = content.replace(",y)", "")
+    content = content.replace("else ifeq ($", "elseif")
+    content = content.replace("ifeq ($", "if")
+    content = content.replace("ifneq ($(", "if(NOT ")
+    content = content.replace("else\n", "else()\n")
+    content = content.replace("endif\n", "endif()\n")
+    content = content.replace("CHIP_CSRCS = ", "set(SRCS ")
+    content = content.replace("CHIP_CSRCS += ", "  list(APPEND SRCS ")
+    content = content.replace("CSRCS = ", "set(SRCS ")
+    content = content.replace("CSRCS += ", "  list(APPEND SRCS ")
+    content = content.replace(".c\n", ".c)\n")
+
+    return content
+
+
+# common for board Makefile and Make.defs
+def convert_board_common(path, content):
+    content = convert_make_conditional(content)
+
+    # remove make specific lines
+    content = "\n".join(line for line in content.split("\n") if "include" not in line)
+    content = "\n".join(line for line in content.split("\n") if "ASRCS" not in line)
+    content = "\n".join(line for line in content.split("\n") if "DEPPATH" not in line)
+    content = "\n".join(line for line in content.split("\n") if "VPATH" not in line)
+    content = "\n".join(line for line in content.split("\n") if "CFLAGS" not in line)
+
+    content += "target_sources(board PRIVATE ${SRCS})\n\n"
+
+    # linker script
+    if os.path.exists(os.path.join(path, "scripts/ld.script")):
```

Comment:
count = 0
    scriptpath = os.path.join(path, "scripts")
    # linker script
    for filename in os.listdir(scriptpath):
        if filename.endswith(".ld") or filename.endswith(".script"):
            print("found linker script ->", filename)
            content += (
                'set_property(GLOBAL PROPERTY LD_SCRIPT "${NUTTX_BOARD_DIR}/scripts/'
                + filename
                + '")\n'
            )
            count += 1
            continue
        else:
            continue

    if count == 0:
        print("ERROR: not found linker script")

Search the directory scripts for files with the extension .ld and .script
and not name.

Reply:
done